### PR TITLE
#22181 move the "Show all databases" option to the main connection page

### LINKED ISSUE #22181: Make 'Show all databases' checkbox easier to find in PostgreSQL settings
PostgreSQL users can't find there databases because they don't see the 'Show all database' checkbox.
We need to make it more clear.
For example, we can add a hint with the link to the main page of PostgreSQL connection settings.

Details: https://github.com/orgs/dbeaver/discussions/22145

To do:
- Show all  —> move to the Main page, near the Database field
- Enable greyed items

### PR DESCRIPTION
**Pull request by LonwoLonwo**
2024-01-29 13_16_00-Window

Please check that the "Show all databases" works as previously with the "Show templates", the "Show non-available" settings and database filters.

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.ext.postgresql.ui/src/org/jkiss/dbeaver/ext/postgresql/PostgresResources.properties [hunk 1]
```diff
@@ -124,7 +124,7 @@ dialog_setting_connection_host = Host
 dialog_setting_connection_cloud_instance = Host/Instance
 dialog_setting_connection_localClient = Local Client
 dialog_setting_connection_nondefaultDatabase = Show all databases
-dialog_setting_connection_nondefaultDatabase_tip = Show all databases in database navigator.\nIf not set then only one database will be visible.\nThis option works in manual connection configuration type only. 
+dialog_setting_connection_nondefaultDatabase_tip = Show all databases in database navigator.\nIf not set then only one database will be visible.\nThis option works in manual connection configuration type only.\nShowing templates or non-available databases will not work with this disabled setting.
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.ext.postgresql.ui/src/org/jkiss/dbeaver/ext/postgresql/PostgresResources.properties]:
```suggestion
dialog_setting_connection_nondefaultDatabase_tip = Show all databases in database navigator.\nIf not set then only one database will be visible.\nThis option works in manual connection configuration type only.\nOptions to show templates or non-available databases will not work with this option disabled.
```

File: plugins/org.jkiss.dbeaver.ext.postgresql.ui/src/org/jkiss/dbeaver/ext/postgresql/ui/PostgreConnectionPage.java [hunk 2]
```diff
@@ -250,6 +256,11 @@ public void loadSettings() {
             }
             dbText.setText(databaseName);
         }
+        if (showNonDefault != null) {
+            showNonDefault.setSelection(
+                CommonUtils.getBoolean(connectionInfo.getProviderProperty(PostgreConstants.PROP_SHOW_NON_DEFAULT_DB),
+                    DBWorkbench.getPlatform().getPreferenceStore().getBoolean(PostgreConstants.PROP_SHOW_NON_DEFAULT_DB)));
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.ext.postgresql.ui/src/org/jkiss/dbeaver/ext/postgresql/ui/PostgreConnectionPage.java]:
```suggestion
            showNonDefault.setSelection(CommonUtils.getBoolean(
                connectionInfo.getProviderProperty(PostgreConstants.PROP_SHOW_NON_DEFAULT_DB),
                DBWorkbench.getPlatform().getPreferenceStore().getBoolean(PostgreConstants.PROP_SHOW_NON_DEFAULT_DB)
            ));
```
Not sure if it's better, but this way it's clearer that these properties are passed in the `CommonUtils.getBoolean` method

File: plugins/org.jkiss.dbeaver.ext.postgresql.ui/src/org/jkiss/dbeaver/ext/postgresql/ui/PostgreConnectionPage.java [hunk 3]
```diff
@@ -158,12 +156,20 @@ public void widgetSelected(SelectionEvent e) {
         dbText = new Text(addrGroup, SWT.BORDER);
         gd = new GridData(GridData.FILL_HORIZONTAL);
         gd.grabExcessHorizontalSpace = true;
-        gd.horizontalSpan = 3;
+        //gd.horizontalSpan = 3;
```
Comment (serjiokov) [inline plugins/org.jkiss.dbeaver.ext.postgresql.ui/src/org/jkiss/dbeaver/ext/postgresql/ui/PostgreConnectionPage.java]: PLS, remove commented line

**Review by E1izabeth — APPROVED** [review]

**Review by ShadelessFox — APPROVED** [review]

**Comment by Matvey16** [comment]
Verified

**Pull request merged**